proof_ipa.go

```diff
@@ -57,12 +58,14 @@ type Proof struct {
 	Cs         []*Point        // commitments, sorted by their path in the tree
 	PoaStems   [][]byte        // stems proving another stem is absent
 	Keys       [][]byte
-	Values     [][]byte
+	PreValues  [][]byte
+	PostValues [][]byte
 }
 
 type SuffixStateDiff struct {
 	Suffix       byte      `json:"suffix"`
 	CurrentValue *[32]byte `json:"currentValue"`
+	NewValue     *[32]byte `json:"new_value"`
```

Comment:
Open question: is there a reason we want a pointer to an array? Feels to me like an unnecessary level of indirection.

Reply:
the problem is that we want the encoding to be `null` if the value doesn't get changed. We could set a slice, as in https://go.dev/play/p/zTnpuXBpY3J, but it means that there could be non-32 bytes values.

Reply:
actually it might be catcheable, but I won't do this in this PR. Created issue #388.